megamek/src/megamek/common/MoveStep.java

```diff
@@ -3356,24 +3364,76 @@ public boolean isMovementPossible(IGame game, Coords src, int srcEl, CachedEntit
                 return false;
             }
         }
+        
+        // Sheer Cliffs, TO p.39
+        // Roads over cliffs cancel the cliff effects for units that move on roads
+        boolean vehicleAffectedByCliff = entity instanceof Tank 
```

Comment:
It looks like a triplication of the 'is this affected by a cliff' code, so yet another argument to consolidate that logic.